### ISSUE TITLE
logging and archiving: discontinuity seen on summertime changes

### ISSUE THREAD

**Issue by FreddieAkeroyd** [ISISComputingGroup/IBEX#8559]
As a user of ibex i would like the logs to function correctly during a clock change. It seems both log plotter and nexus files showed overplotting of values when clocks changed in october. 

![image](https://github.com/user-attachments/assets/242f9dc7-f411-460e-8e42-91ba406d95d7)

it looks like values are archived in mysql in local time rather than UTC

![image](https://github.com/user-attachments/assets/0e8c6014-cad4-4839-8646-bdf61f1e6986)

## Acceptance criteria

- [ ] values are logged in mysql via archive engine so thay are displayed correctly

## notes

- various places to look e.g. jdbc connection parameters
https://github.com/ISISComputingGroup/EPICS-CSS/blob/master/ArchiveEngine/mysql_settings_inst.ini#L2 maybe we inconsistent for how we format the time value we inset in the archive engine, or maybe the PV timestamp if getting interpreted  incorrectly
- now that we are on winter time not sure how easy it is to test, may need to do on a computer with a forced timezone change?

## Planning
27/08/2025 -00:03:30
28/11/2024 - 00:14:00
31/10/2024 -  104:00

**Comment by Tom-Willemsen** [comment]
defer

**Comment by FreddieAkeroyd** [comment]
Looking at values currently being archived it would seem that BST values are going in as UTC in the table. Assuming this is true for post BST, then it suggests the issue is with retrieval rather than archiving? I created a test:    
```
CREATE TABLE sample_test ( smpl_time TIMESTAMP NOT NULL );
SET time_zone = '+00:00';
INSERT INTO sample_test VALUES ('2025-10-26 00:10:00'),('2025-10-26 01:10:00'),('2025-10-26 02:10:00'),('2025-10-26 03:10:00');
select * from sample_test;
set time_zone='SYSTEM';
select * from sample_test;
```
which gives
```
mysql> select * from sample_test;
+---------------------+
| smpl_time           |
+---------------------+
| 2025-10-26 00:10:00 |
| 2025-10-26 01:10:00 |
| 2025-10-26 02:10:00 |
| 2025-10-26 03:10:00 |
+---------------------+
4 rows in set (0.13 sec)

mysql> SET time_zone = 'SYSTEM';
Query OK, 0 rows affected (0.00 sec)

mysql> select * from sample_test;
+---------------------+
| smpl_time           |
+---------------------+
| 2025-10-26 01:10:00 |
| 2025-10-26 01:10:00 |
| 2025-10-26 02:10:00 |
| 2025-10-26 03:10:00 |
+---------------------+
4 rows in set (0.00 sec)
```
so  `2025-10-26 01:10:00` is now duplicated. I think this confirms what @Chsudeepta suggested in a conversation I had with him that the plotting behaviour is expected. The query is converting back in local time after re-applying DST. The problem is the plot axis can't cope with this as it can only show one timezone. The issue would go away i think if the plot used your current timezone offset and not your timezone full i.e. used '+00:00' and not 'Europe/London', in this case. That was rather than converting 00:10:00 to 01:10:00 as it is in BST at the time,  it would leave it as 00:10:00 as the current user timezone (as opposed to one in place at value recording time) is not BST. This stops a discontinuity, but means all historic values are not in local time which may be more confusing than a discontinuity that looks to be just visualisation and not an archiving issue. So is it best if we leave this as a "feature" and "won't fix"? Maybe we could add an option to the graph to force a totally UTC plot if somebody did want to plot across the clock change?

**Comment by Chsudeepta** [comment]
Yes, it might be better to express the requirements more explicitly in a new ticket if we need to change behaviour

**Comment by Tom-Willemsen** [comment]
Not planned at this moment - can reopen if decision changes later.
